common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -256,6 +258,10 @@ object DotcomRenderingDataModel {
         .getOrElse(blocks.body.fold(Seq.empty[APIBlock])(_.filter(_.attributes.pinned.contains(true))))
         .headOption
 
+    val mostRecentBlockId = page.currentPage.pagination.flatMap(paginationInfo =>
+      paginationInfo.newest.flatMap(_.blocks.headOption.map(_.id)),
```

Comment:
Is the `newest` property  always available? I ask because I know the client side `pagination` object is sometimes missing this value based on what page you're on. Of course, this version of the data could act differently which is fine but thought I'd check

Reply:
It isn't always there no, but it should always be there if you're on an older page, which is when you need mostRecentBlockId. Do we need mostRecentBlockId to be available on the first page?

Reply:
I've changed to use the firstPage blocks, I think you're right it will be less confusing